src/main/java/com/boomi/flow/services/boomi/mdh/match/MatchEntityRepository.java

```diff
@@ -80,15 +79,23 @@ public MatchEntityRepository(MdhClient mdhClient) {
     private BatchUpdateRequest.Entity createUpdateEntity(Universe universe, MObject entity) {
         // Map all the properties to fields, except our "internal" ones
         Map<String, Object> fields = mObjectToMap(entity);
-
-        fields.put(universe.getIdField(), entity.getExternalId());
+        fields.put(universe.getIdField(), extractFieldIdValue(entity));
 
         return new BatchUpdateRequest.Entity()
                 .setOp(null)
                 .setName(universe.getLayout().getModel().getName())
                 .setFields(fields);
     }
 
+    private String extractFieldIdValue(MObject object) {
+        return object.getProperties()
+                .stream()
+                .filter(p -> GoldenRecordConstants.ENTITY_ID_FIELD.equals(p.getDeveloperName()))
+                .findFirst()
+                .map(Property::getContentValue)
+                .orElse("");
```

Comment:
Is an empty string still a valid ID? Or should this be `null` if it's for entities that don't have an ID yet?

Reply:
You are right I have changed this method in the pull request https://github.com/manywho/service-mdh/pull/21  to 
````extractFieldIdValueOrRandomGenerate````
The API for matching entities doesn't support empty or null values.